frontend/src/actions/authActions.ts

```diff
@@ -41,7 +48,14 @@ export const signUserUp = (payload: IUserDetails) =>
 
     try {
       const result = await authApiClient.register(payload)
-      localStorage.setItem('token', result.token.access)
+      const tokenStorage = new TokenStorage()
+      tokenStorage.saveToken(result.token.access)
+      tokenStorage.saveRefreshToken(result.token.refresh)
+      const wholeToken: IToken  = jwtDecode(result.token.access)
+      result.id = wholeToken.payload.user_id
```

Comment:
Modification of `result` is not good approach. Prefer creating new objects instead of modifying, unless there is serious reason to use mutation over creating new (for instance - performance issue)/
As I can see from here, `result` is used only in `setUserAction` which is consume `IUser` interface. Please, instead of providing `result`, pass an object with only required fields. Same for action down below.

Reply:
fixed for both login and register